Missing TypeScript type declarations for d3-cloud and jsonpath packages

## Issue Summary

TypeScript errors due to missing type declarations for third-party packages `d3-cloud` and `jsonpath`.

## TypeScript Errors

```
app/collections/_components/word-cloud.tsx(98,32): error TS7016: Could not find a declaration file for module 'd3-cloud'. '/Users/huohao/.openclaw/workspace/ossinsight/node_modules/.pnpm/d3-cloud@1.2.9/node_modules/d3-cloud/build/d3.layout.cloud.js' implicitly has an 'any' type.

lib/charts-core/datasource/api.ts(4,16): error TS7016: Could not find a declaration file for module 'jsonpath/jsonpath.js'. '/Users/huohao/.openclaw/workspace/ossinsight/node_modules/.pnpm/jsonpath@1.3.0/node_modules/jsonpath/jsonpath.js' implicitly has an 'any' type.
```

## Impact

- **Type safety**: These packages are typed as `any`, losing type checking benefits
- **Developer experience**: No IntelliSense or autocomplete for these packages
- **Code quality**: Potential runtime errors may not be caught at compile time

## Files Affected

- `apps/web/app/collections/_components/word-cloud.tsx` (uses `d3-cloud`)
- `apps/web/lib/charts-core/datasource/api.ts` (uses `jsonpath`)

## Suggested Fix

Install the type declaration packages:

```bash
# For d3-cloud
pnpm add -D @types/d3-cloud

# For jsonpath
pnpm add -D @types/jsonpath
```

If `@types/jsonpath` does not exist, add a type declaration file:

```typescript
// apps/web/types/jsonpath.d.ts
declare module 'jsonpath/jsonpath.js' {
  export function query(obj: any, path: string): any[];
  export function paths(obj: any, path: string): string[];
  export function nodes(obj: any, path: string): any[];
}
```

## Priority

**Low** - Does not affect runtime, but improves type safety and developer experience

## References

- Related to #2344 (TypeScript errors masked by ignoreBuildErrors)

--- a/apps/web/package.json
+++ b/apps/web/package.json
@@ -90,6 +90,7 @@
     "@svgr/webpack": "^8.1.0",
     "@tailwindcss/typography": "^0.5.10",
     "@types/d3-cloud": "^1.2.9",
+    "@types/jsonpath": "^0.2.4",
     "@types/lodash": "^4.14.199",
     "@types/luxon": "^3.3.2",
     "@types/mdast": "^4.0.0",

